compiler/rustc_abi/src/lib.rs

```diff
@@ -253,6 +253,98 @@ impl TargetDataLayout {
         // That is, use the size, rounded up to a power of 2.
         AbiAndPrefAlign::new(Align::from_bytes(vec_size.bytes().next_power_of_two()).unwrap())
     }
+
+    pub fn parse_from_llvm_datalayout_string<'a>(
```

Comment:
Can you move this above the other methods? I'm basing that [on what it used to look like](https://github.com/rust-lang/rust/8876ca3dd46b99fe7e6ad937f11493d37996231e/compiler/rustc_target/src/abi/mod.rs).

You may also need to add a doc comment saying that this does not set `c_enum_min_size` because there's no way to know what the value should be at that point.

Alternatively, you could take a 4-field `struct` that contains a subset of `Target`'s fields (`data_layout`, `endian`, `pointer_width`, `c_enum_min_bits`).

I'm not too happy with `c_enum_min_bits` overall, because `TargetDataLayout` was meant to contain all the information LLVM has access to, but I understand the desire to cache some of it in a convenient place.